ddtrace/tracer/span_test.go

```diff
@@ -83,14 +83,21 @@ func TestNilSpan(t *testing.T) {
 	assertions.Nil(ctx)
 	assertions.Equal(TraceIDZero, ctx.TraceID())
 	assertions.Equal([16]byte(emptyTraceID), ctx.TraceIDBytes())
+	assertions.Equal(uint64(0), ctx.TraceIDLower())
 	assertions.Equal(uint64(0), ctx.SpanID())
 	sp, ok := ctx.SamplingPriority()
 	assertions.Equal(0, sp)
 	assertions.Equal(false, ok)
 	// calls on nil span should be no-op
 	assertions.Nil(span.Root())
 	span.SetBaggageItem("key", "value")
+	if v := span.BaggageItem("key"); v != "" {
+		t.Errorf("expected empty string, got %s", v)
+	}
 	span.SetTag("key", "value")
+	if v := span.Tag("key"); v != nil {
+		t.Errorf("expected nil, got %s", v)
+	}
 	span.SetUser("user")
 	assertions.Nil(span.StartChild("child"))
 	span.Finish()
```

Comment:
Should we also check the behavior when setting tags with `tracer.StartSpan` + `tracer.StartSpanOption`? 🤔 

```go
span := tracer.StartSpan("operation-name", tracer.ServiceName("service"), tracer.Tag("tag", "value"))
```